### PR TITLE
feat(python): combine load_version/load_with_datetime into `load_as_version`

### PR DESCRIPTION
**Pull request by ion-elgreco**
# Description
Combines the two functions into one.

# Related Issue(s)
- closes https://github.com/delta-io/delta-rs/issues/1910
- closes https://github.com/delta-io/delta-rs/issues/1967

### CODE REVIEW

File: python/deltalake/table.py [hunk 1]
```diff
@@ -453,20 +453,68 @@ def file_uris(
...
+            ```
+            datetime(2023,1,1)
+            ```
+
+            **Use a datetime in string format**
+            ```
+            "2018-01-26T18:30:09Z"
+            "2018-12-19T16:39:57-08:00"
+            "2018-01-26T18:30:09.453+00:00"
+            ```
```
Comment (roeap) [inline python/deltalake/table.py:480]: should these examples show the actual function invocaton?

**Review by roeap — COMMENTED** [review]
thanks for this refactor.

somehow I am not a fan of the name `load_to`, but also don't have a better one ready.

**Comment by roeap** [comment]
maybe an opportunity to fix #1967 as well?

File: python/deltalake/table.py (hunk 1 above)
Comment (ion-elgreco) [reply python/deltalake/table.py:480]: Yeah, that would be better

**Comment by ion-elgreco** [comment]
> thanks for this refactor.
> 
> somehow I am not a fan of the name `load_to`, but also don't have a better one ready.

@roeap maybe we can just name it `load` instead?

**Comment by roeap** [comment]
I try to always see if it works as a short sentence ... "load [table] to version" just somehow hit my ear wrong 😆. 

Would "load [table] as version" work? `load` alone may be confusion as well, as this may be misconsrued as the table still needs to be loaded?

**Comment by ion-elgreco** [comment]
@roeap Yeah when you read it like that it doesn't sound right haha. Hmm, that would translate into load_as_version or load_asof?

**Comment by roeap** [comment]
Not sure myself, but what would you think about `at` ... 

```
table.at(version)
```

**Comment by ion-elgreco** [comment]
@roeap I think `at` is to unclear. Would the previously mentioned `load_state` not be more clear? Since it's doing that?

**Comment by ion-elgreco** [comment]
> maybe an opportunity to fix #1967 as well?

@roeap fixing this, causes now one of the restore test in Rusts to fail, not sure what to do here because all python tests pass fine.

**Review by roeap — APPROVED** [review]
👍

**Pull request merged**